src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
the constants are for use in your own code *if you want them*. But they are optional.

But if something is out of sync, we should better use the status code here, not the constant name with the wrong status code, as the HTTP spec cares about the status code, not about the name we gave to the constant in our code (the HTTP stack never sees this constant anyway).
So I don't see your point.

And for someone knowing what meaning status codes have, I find ``array(301, 302, 307, 308)`` much more readable than the list of constant names, having to remember what their mapping to actual status codes are.